src/engine/StringMapping.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>
+//          Robin Textor-Falconi <textorr@cs.uni-freiburg.de>
+
+#ifndef QLEVER_SRC_ENGINE_STRING_MAPPING_H
+#define QLEVER_SRC_ENGINE_STRING_MAPPING_H
+
+#include <string>
+#include <vector>
+
+#include "global/Id.h"
+#include "util/HashMap.h"
+
+// Forward declaration
+class Index;
+
+namespace qlever::binary_export {
+// Helper struct to temporarily store strings and aggregate them to avoid
+// sending the same strings over and over again.
+struct StringMapping {
+  // Store the actual mapping from an id to the corresponding insertion order.
+  // (The first newly inserted string will get id 0, the second id 1, and so
+  // on.)
+  // TODO<RobinTF, joka921> consider mapping from `Id::T` instead of `Id` to get
+  // a cheaper lookup.
+  ad_utility::HashMap<Id, uint64_t> stringMapping_;
+  uint64_t numProcessedRows_ = 0;
+
+  // Serialize the internal datastructure to a string and clear it.
+  std::vector<std::string> flush(const Index& index);
```

Comment:
The comment is off (we now return a vecotr of strings), and also document, how the mapping is reflected here (the index in the result is the value of the hash map basically.